metaflow/datatools/s3.py

```diff
@@ -604,17 +780,24 @@ def _head():
     def get(self, key=None, return_missing=False, return_info=True):
         """
         Get a single object from S3.
-        Args:
-            key: (optional) a suffix identifying the object. Can also be
-                 an object containing the properties `key`, `offset` and
-                 `length` to specify a range query. `S3GetObject` is such an object.
-            return_missing: (optional, default False) if set to True, do
-                            not raise an exception for a missing key but
-                            return it as an S3Object with .exists == False.
-            return_info: (optional, default True) if set to True, fetch the
-                         content-type and user metadata associated with the object.
-        Returns:
-            an S3Object corresponding to the object requested.
+
+        Parameters
+        ----------
+        key : str or `S3GetObject`
```

Comment:
when we had first introduced S3GetObject, you were hesitant to restrict it to just this thus the doc was more general. I think this is clearer but just bringing it up in case you want to leave it a bit more generic.

nit: an `S3GetObject` which defines the key as well as a range of data to download.